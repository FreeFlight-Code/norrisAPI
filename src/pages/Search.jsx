import React, { Component } from "react";
import {connect} from 'react-redux';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: "",
      jokes: []
    };
  }

  handleinputChange(val){
    this.setState({
      searchTerm: val
    })
  }

  handleSubmit(){
    const {searchTerm} = this.state;
    fetch(`https://api.chucknorris.io/jokes/search?query=${searchTerm}`)
      .then(res => res.json())
      .then(obj => {
        let jokesArray = obj.result;
        this.setState({ 
          searchTerm: "",
          jokes: jokesArray
        })
      });
  }

  truncJoke(string, charLimit){
    return string.slice(0, charLimit + 1) + "..."
  }

  renderResults(){
    const {jokes} = this.state;
    if(jokes && jokes.length > 0){
      return jokes.map((joke, i)=>{
        let shortJoke = this.truncJoke(joke.value, 50);
        return (
          <div
            onClick={e => this.props.dispatch({ type: "DISPLAY_MODAL", message: joke.value, messageType: "info" })}
            key={`joke-${i}`}
          >
            {shortJoke}
          </div>
        );
      })
    }
  }

  render() {
    return (
      <div className="page search">
        <label htmlFor="searchBar">Search Bar</label>
        <input value={this.state.searchTerm} id="searchBar" type="text" onChange={e=>{this.handleinputChange(e.target.value)}} placeholder="Search Chuck Norris Joke by Key Word"></input>
        <button className="blue" onClick={e=>{this.handleSubmit()}}>Submit</button>
        <div className="results">
          <h3>Results</h3>
          {this.renderResults()}
        </div>
      </div>
    )
  }
}

export default connect()(Search);