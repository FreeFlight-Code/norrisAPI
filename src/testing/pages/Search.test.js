import React from "react";
import Search, {SearchResults} from "../../pages/Search";
import renderer from "react-test-renderer";
import { fireEvent } from "@testing-library/react";
import {render} from '../test-utils';


describe(" The Search Page... ", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Results renders correctly", () => {
    const tree = renderer.create(<SearchResults />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(' checks input ', () => {
		const { getByTestId } = render(<Search />);
    const input = getByTestId("search-input");
    fireEvent.change(input, { target: { value: "jokes" } });
    expect(input.value).toBe("jokes");
  })
  it(' checks submit ', () => {
		const { getByTestId } = render(<Search />);
    const input = getByTestId("submit-input");
    fireEvent.click(input);
  })
});
