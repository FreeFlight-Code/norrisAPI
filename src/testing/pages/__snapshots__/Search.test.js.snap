// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` The Search Page...  Results renders correctly 1`] = `null`;

exports[` The Search Page...  renders correctly 1`] = `
<div
  className="page search"
  style={
    Object {
      "background": "linear-gradient(#dddcdc 10%, #b3b2b2)",
      "padding": "10vh 3vw",
    }
  }
>
  <label
    htmlFor="searchBar"
  >
    Search Bar
  </label>
  <input
    data-testid="search-input"
    id="searchBar"
    onChange={[Function]}
    placeholder="Search Chuck Norris Joke by Key Word"
    style={
      Object {
        "padding": "10px 15px",
      }
    }
    type="text"
    value=""
  />
  <button
    className="blue"
    data-testid="submit-input"
    onClick={[Function]}
  >
    Submit
  </button>
  <div
    className="results"
  >
    <h3>
      Results
    </h3>
  </div>
</div>
`;
